core/src/main/java/io/questdb/griffin/engine/table/ShowCreateMatViewRecordCursorFactory.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("show_create_materialized_view");
+        sink.meta("of").val(tableToken.getTableName());
+    }
+
+    public static class ShowCreateMatViewCursor implements NoRandomAccessRecordCursor {
+        protected final Utf8StringSink sink = new Utf8StringSink();
+        private final ShowCreateMatViewRecord record = new ShowCreateMatViewRecord();
+        protected SqlExecutionContext executionContext;
+        protected MatViewDefinition matViewDefinition;
+        protected CairoTable table;
+        private boolean hasRun;
+        private TableToken tableToken;
+
+        @Override
+        public void close() {
+            sink.clear();
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (!hasRun) {
+                sink.clear();
+                final CairoConfiguration config = executionContext.getCairoEngine().getConfiguration();
+                showCreateMatView(config);
+                hasRun = true;
+                return true;
+            }
+            return false;
+        }
+
+        public ShowCreateMatViewCursor of(
+                SqlExecutionContext executionContext,
+                TableToken tableToken,
+                int tokenPosition
+        ) throws SqlException {
+            this.tableToken = tableToken;
+            this.executionContext = executionContext;
+            try (MetadataCacheReader metadataRO = executionContext.getCairoEngine().getMetadataCache().readLock()) {
+                this.table = metadataRO.getTable(tableToken);
+                if (this.table == null) {
+                    throw SqlException.$(tokenPosition, "table does not exist [table=")
+                            .put(tableToken.getTableName()).put(']');
+                } else if (!tableToken.equals(table.getTableToken())) {
+                    throw TableReferenceOutOfDateException.of(this.tableToken);
+                }
+            }
+
+            if (!tableToken.isMatView()) {
+                throw SqlException.$(0, "materialized view expected, got table");
+            }
+
+            CairoConfiguration configuration = executionContext.getCairoEngine().getConfiguration();
+            Path path = Path.getThreadLocal(configuration.getDbRoot());
+            final int pathLen = path.size();
+            try (BlockFileReader reader = new BlockFileReader(configuration)) {
+                try {
+                    this.matViewDefinition = MatViewDefinition.readFrom(
+                            reader,
+                            path,
+                            pathLen,
+                            tableToken
+                    );
+                } catch (CairoException e) {
+                    throw SqlException.$(0, "could not read materialized view definition [table=").put(tableToken)
```

Comment:
there is an opportunity to have non-zero error position here, table name is not at position 0